dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/programrule/ProgramRuleIntegrationTest.java

```diff
@@ -179,7 +188,7 @@ void testImportEventInProgramStageSuccessWithWarningRaised()
         assertEquals( 3, warningReports.size() );
         assertEquals( TrackerErrorCode.E1308, warningReports.get( 0 ).getWarningCode() );
         assertEquals(
-            "Generated by program rule (`ProgramRule`) - DataElement `DATAEL00001` is being replaced in event `EVENT123456`",
+            "Generated by program rule (`ProgramRulC`) - DataElement `DATAEL00001` is being replaced in event `EVENT123456`",
```

Comment:
why is that change necessary?

Reply:
We need to talk about this...
There are a few places that are quite difficult to test because of Hibernate Session.
This particular one has something to do with the fact that Hibernate cache doesn't work well if we retrieve a filtered list of program rule actions linked to a program rule.